src/diff/index.js

```diff
@@ -161,16 +166,17 @@ export function diff(parentDom, newVNode, oldVNode, context, isSvg, excessDomChi
 	return newVNode._dom;
 }
 
-export function commitRoot(mounts, root) {
-	let c;
-	while ((c = mounts.pop())) {
+export function commitRoot(commitQueue, root) {
+	commitQueue.some(c => {
 		try {
-			c.componentDidMount();
+			commitQueue = c._renderCallbacks;
+			c._renderCallbacks = [];
+			commitQueue.some(cb => { cb.call(c); });
```

Comment:
@developit, it seems back in #648, you [expressed perf concerns about a general callback queue](https://github.com/preactjs/preact/issues/648#issuecomment-295326760), which I think is what I've implemented here. Any concerns with this implementation?